pkg/controller/certificaterequest/certificaterequest_controller.go

```diff
@@ -145,18 +146,41 @@ func (r *ReconcileCertificateRequest) Reconcile(request reconcile.Request) (reco
 	}
 
 	certificateSecret := newSecret(cr)
+	platformSecretName := cr.Spec.PlatformSecrets.AWS.Credentials.Name
 
 	// Set CertificateRequest cr as the owner and controller
 	if err := controllerutil.SetControllerReference(cr, certificateSecret, r.scheme); err != nil {
 		reqLogger.Error(err, err.Error())
 		return reconcile.Result{}, err
 	}
 
-	found := &corev1.Secret{}
+	// Ensure that platform Secret exists.
+	awscreds := &corev1.Secret{}
+	credserr := r.client.Get(context.TODO(), types.NamespacedName{Name: platformSecretName, Namespace: request.Namespace}, awscreds)
+	if credserr != nil {
+		reqLogger.Info("platformSecrets were not found. Unable to search for certificates in cloud provider platform")
+		return reconcile.Result{}, credserr
+	}
 
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: certificateSecret.Name, Namespace: certificateSecret.Namespace}, found)
+	// Ensure that platform Secret can authenticate to AWS.
```

Comment:
I think it might be best if this logic block was moved behind the awsclient and abstracted away. We could call it awsclient.TestAuth() or something like that.